Add issue-wanted idea (#1)
kowainik/summer-of-haskell: content/ideas/issue-wanted.md

---
title: "issue-wanted: Web app for discovering potential contributions to the Haskell ecosystem"
---

Haskell ecosystem can be improved in many ways. There are a lot of libraries
with tons of issues. But it's not that easy to discover potential problems you
would like to work on. The goal of [issue-wanted](https://github.com/kowainik/issue-wanted)
is to help find GitHub issues using GitHub labels and various metadata fields
from `.cabal` files (like `category`).

Project contains several parts:

1. Haskell process that syncs information about Haskell repositores at GitHub.
2. Haskell backend that provides REST API for fetched information.
3. Elm frontend that displays this information.

This project is a great way for students to work on real-world Haskell project
using modern libraries and approaches.

**Potential Mentors**: Dmitrii Kovanikov, Veronika Romashkina

**Difficulty**: Intermediate

> Mentors take care of Elm frontend and deployment. Student can work only on

Author: I wanted to say that if student want, he/she also can work on frontend and DevOps tasks :slightly_smiling_face: Saying that we can take care of this seems good enough to me :+1:

> Haskell part. The choice of main libraries for the project is the following:

Author: I think it's better to say `suggested` as you proposed :+1: Though, I really don't want to use `spock` or something like that. And in that case I can't really mentor the project.

* `servant` for writing REST API servers and clients in Haskell
* PostgreSQL as database using `postgresql-simple` Haskell library
* JSON as a communication format between backend and frontend, `aeson` Haskell library

> * `async` for writing concurrent code

Author: Not sure... We haven't write this library yet :slightly_smiling_face:

---
title: "issue-wanted: Web app for discovering potential contributions to the Haskell ecosystem"
---

> Haskell ecosystem can be improved in many ways. There many libraries of different categories

Reviewer: ```suggestion
Haskell ecosystem can be improved in many ways. There are many libraries of different categories
```

with tons of issues — starting with low-hanging fruits to expert level problems. However, there is no easy way to discover potential problems you

> would like to work on. The goal of [issue-wanted](https://github.com/kowainik/issue-wanted)

Reviewer: Work on or work at? :thinking:

Author: Looks like they both equivalent:

* https://ell.stackexchange.com/a/98307

But for me `work at` usually means some place, like `work at Kowainik`

> is to help find GitHub issues using GitHub labels and various metadata fields

Reviewer: "find project issues"?

from `.cabal` files (like `category**).

The project contains several parts:

1. Haskell process that syncs information about Haskell repositories at GitHub.
2. Haskell backend that provides REST API for fetched information.
3. Elm frontend that displays this information.

The benefits of this project are the following:

> 1. Easier for beginners and experts to find issues they can solve.

Reviewer: With `issue-wanted` it's more accessible fo beginners to find interesting issues they can solve.

> 2. Average maintenance of Haskell libraries will be improved as consequence: if

Reviewer: ```suggestion
2. Average maintenance of Haskell libraries will be improved as a consequence: if
```

>    people want more open-source contributors, they will provide better meta

Reviewer: probably "project owners" instead of "people"?

>    people want more open-source contributors, they will provide better meta

Reviewer: They *could* provide..

>    information to their packages and this actuall will pay off.

Reviewer: ```suggestion
   information to their packages, and this actuall will pay off.
```

> 

Reviewer: typo in `actually`, also better not to use *will*, you can say this pays off (?)

I would add that any project could become more popular and gather more attention, but not sure how to tell this correctly..

> This project is an excellent way for students to work on real-world Haskell project

Reviewer: ```suggestion
This project is an excellent way for students to work on the real-world Haskell project
```

using modern libraries and approaches.

**What is expected from student**

> 1. Basic knowledge of Haskell programming language: `newtype`, typeclasses,

Reviewer: knowledge of Haskell programming language *concepts*/*definitions*/*something else*?

   `Monad`s, ability to write code with `Reader` or `State` monad.
2. Very basic knowledge of SQL. At least student should not be afraid to learn

>    and use SQL, though it's not expected to have complicated queires in this

Reviewer: ```suggestion
   and use SQL, though it's not expected to have any complex queries in this
```

   project.
3. Patience. Reading GitHub API documentation and testing integration of
   multiple systems might be frustrating sometimes.

> **What student will get after this project**

Reviewer: Will or would?

1. Experience of writing REST API web services in Haskell.

> 2. Familarity with intermediate Haskell features like monad transformers and

Reviewer: ```suggestion
2. Familiarity with intermediate Haskell features like monad transformers and
```

   type-level computations.
3. Knowledge of modern Haskell programming patterns and idioms.

**Potential Mentors***: Dmitrii Kovanikov, Veronika Romashkina

**Difficulty**: Intermediate

Mentors can take care of Elm frontend and deployment. The student can work only

> on the Haskell part (but also can work on frontend and DevOps tasks). Suggested

Reviewer: Probably better to say, that it's not restricted to Haskell only, if they have any interest in other details of the projects it's free to experiment with. Or something in better word, I hope you got the idea I wanted to point out to
